MadDonkeySoftware/corp-hq
# Chq 34 - Server side validation dotnet-ified

### Story
https://maddonkeysoftware.myjetbrains.com/youtrack/issue/CHQ-34

### Summary
* added ModelState.IsValid
*  added ModelStateExtensions to extract errors from ModelStateDictionary as ModelState.Errors()

### Testing
* Start the UI and API projects with mongo running
* Navigate to the Registration Page
* Click Submit with no data present or missing individual fields
* You will see the validation returned from the server at the top of the screen

NOTE: I will fix the screen layout with client side validation from https://maddonkeysoftware.myjetbrains.com/youtrack/issue/CHQ-25
